dm/syncer/data_validator.go

```diff
@@ -571,6 +570,7 @@ func (v *DataValidator) Stage() pb.Stage {
 
 func (v *DataValidator) startValidateWorkers() {
 	v.wg.Add(v.workerCnt)
+	v.workers = make([]*validateWorker, v.workerCnt)
```

Comment:
which line panic,  can't understand why this fixes the problem, could you please explain more?

Reply:
ptal https://github.com/pingcap/tiflow/pull/5181#discussion_r852549354

Reply:
v.workers has already inited here
https://github.com/pingcap/tiflow/blob/56ccdf2178468331687ae2f07e83e38111811b27/dm/syncer/data_validator.go#L174

it panics here? not checks worker:
https://github.com/pingcap/tiflow/blob/56ccdf2178468331687ae2f07e83e38111811b27/dm/syncer/validator_checkpoint.go#L191

maybe we should init this field when create workers

Reply:
> v.workers has already inited here
> 
> https://github.com/pingcap/tiflow/blob/56ccdf2178468331687ae2f07e83e38111811b27/dm/syncer/data_validator.go#L174

Yes, and I removed this to put off initializing the workers array till we start validating. 

https://github.com/pingcap/tiflow/blob/56ccdf2178468331687ae2f07e83e38111811b27/dm/syncer/validator_checkpoint.go#L191
> 
> maybe we should init this field when create workers

It's not because we didn't initialize worker's field but merely the worker itself is nil (because we just initialized the workers' array but the workers inside were nil before starting).